### PR TITLE
byedpi: init at 0.13.1

### PR DESCRIPTION
**Pull request by AveryanAlex**
## Init byedpi at 0.13.1

https://github.com/hufrea/byedpi

Anti-censorship proxy server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/by/byedpi/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
```
Comment (iblech) [inline pkgs/by-name/by/byedpi/package.nix:2]: I think our standard style would be as follows:

```
{ lib
, stdenv
, fetchFromGitHub
, nix-update-script
}:
```

File: pkgs/by-name/by/byedpi/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  installPhase = ''
+    install -Dm755 ciadpi $out/bin/ciadpi
+  '';
+
+  strictDeps = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "SOCKS proxy server implementing some of DPI bypass methods";
```
Comment (iblech) [inline pkgs/by-name/by/byedpi/package.nix:27]: s/some of/some/

Thank you for your pull request! :wave:

File: pkgs/by-name/by/byedpi/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  nix-update-script,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "byedpi";
+  version = "0.12";
+
+  src = fetchFromGitHub {
+    owner = "hufrea";
+    repo = "byedpi";
+    rev = "refs/tags/v${finalAttrs.version}";
```
Comment (iblech) [inline pkgs/by-name/by/byedpi/package.nix:14]: I believe that you can shorten this to just `"v${finalAttrs.version}"`, i.e. no need for `refs/tags/`.

File: pkgs/by-name/by/byedpi/package.nix (hunk 1 above)
Comment (averyanalex) [reply pkgs/by-name/by/byedpi/package.nix:2]: https://github.com/NixOS/rfcs/pull/166

Comment (averyanalex) [reply pkgs/by-name/by/byedpi/package.nix:2]: Btw I used `nixfmt-rfc-style` to format code

File: pkgs/by-name/by/byedpi/package.nix (hunk 2 above)
Comment (averyanalex) [reply pkgs/by-name/by/byedpi/package.nix:27]: Thanks! But maybe completely rewrite description? Like "SOCKS5 proxy server that bypasses dpi-based censorship"?

File: pkgs/by-name/by/byedpi/package.nix (hunk 1 above)
Comment (iblech) [reply pkgs/by-name/by/byedpi/package.nix:2]: Oh, I was not aware of this change, thank you for bringing it to my attention! I stand corrected then.

File: pkgs/by-name/by/byedpi/package.nix (hunk 3 above)
Comment (averyanalex) [reply pkgs/by-name/by/byedpi/package.nix:14]: I checked some PRs and packages - some uses "v{version}", some "refs/tags/v{version}". If the long version is used, then there are some reasons for this (i.e. explicitness)? I couldn't find any recommendations about it.

File: pkgs/by-name/by/byedpi/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-Sf5Ik8+9nKhum4/faGf44Y/gQggldyRsFUVmd9XPKOA=";
+  };
+
+  installPhase = ''
+    install -Dm755 ciadpi $out/bin/ciadpi
```
Comment (Aleksanaa) [inline pkgs/by-name/by/byedpi/package.nix:19]:
```suggestion
    runHook preInstall
    install -Dm755 ciadpi $out/bin/ciadpi
    runHook postInstall
```

File: pkgs/by-name/by/byedpi/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "SOCKS proxy server implementing some DPI bypass methods";
+    homepage = "https://github.com/hufrea/byedpi";
+    changelog = "https://github.com/hufrea/byedpi/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ averyanalex ];
+    platforms = lib.platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/by/byedpi/package.nix:32]:
```suggestion
    platforms = with lib.platforms; linux ++ windows;
```

https://github.com/hufrea/byedpi/issues/17

The macros also only mentioned linux and windows.

File: pkgs/by-name/by/byedpi/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/by/byedpi/package.nix:14]: To differentiate with branches. Although imo fetching a branch is a stupid idea

File: pkgs/by-name/by/byedpi/package.nix (hunk 5 above)
Comment (averyanalex) [reply pkgs/by-name/by/byedpi/package.nix]: Yes, looks like it only supports linux and windows, thanks
